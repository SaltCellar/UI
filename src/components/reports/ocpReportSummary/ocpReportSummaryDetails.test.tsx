import { shallow } from 'enzyme';
import React from 'react';
import {
  OcpReportSummaryDetails,
  OcpReportSummaryDetailsProps,
} from './ocpReportSummaryDetails';

const props: OcpReportSummaryDetailsProps = {
  report: { data: [], meta: { total: { cost: { value: 100, units: 'USD' } } } },
  label: 'label',
  formatValue: jest.fn(() => 'formatedValue'),
};

test('report total is formated', () => {
  const view = shallow(<OcpReportSummaryDetails {...props} />);
  expect(props.formatValue).toBeCalledWith(
    props.report.meta.total.cost.value,
    props.report.meta.total.cost.units,
    props.formatOptions
  );
  expect(view).toMatchSnapshot();
});

test('defaults value if report is not present', () => {
  const view = shallow(<OcpReportSummaryDetails {...props} report={null} />);
  expect(props.formatValue).not.toBeCalled();
  expect(view).toMatchSnapshot();
});

test('defaults value if report.meta is not present', () => {
  const view = shallow(
    <OcpReportSummaryDetails
      {...props}
      report={{ ...props.report, meta: null }}
    />
  );
  expect(props.formatValue).not.toBeCalled();
  expect(view).toMatchSnapshot();
});
