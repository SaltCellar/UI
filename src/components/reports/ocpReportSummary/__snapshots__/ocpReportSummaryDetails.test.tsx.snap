// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defaults value if report is not present 1`] = `
<Fragment>
  <div
    className="css-1br38yu"
  >
    <div
      className="css-qvzirm"
    >
      ----
      <div
        className="css-5bzg50"
      >
        <div>
          label
        </div>
      </div>
    </div>
  </div>
  <div
    className="css-1br38yu"
  />
</Fragment>
`;

exports[`defaults value if report.meta is not present 1`] = `
<Fragment>
  <div
    className="css-1br38yu"
  >
    <div
      className="css-qvzirm"
    >
      ----
      <div
        className="css-5bzg50"
      >
        <div>
          label
        </div>
      </div>
    </div>
  </div>
  <div
    className="css-1br38yu"
  />
</Fragment>
`;

exports[`report total is formated 1`] = `
<Fragment>
  <div
    className="css-1br38yu"
  >
    <div
      className="css-qvzirm"
    >
      formatedValue
      <div
        className="css-5bzg50"
      >
        <div>
          label
        </div>
      </div>
    </div>
  </div>
  <div
    className="css-1br38yu"
  />
</Fragment>
`;
